File: SwiftAssignments/SwiftAssignments/Questions/K03.swift
import Foundation

class K03 : Question {
    func question1() {
        print("年齢を入力してください。＞")
        let age = readLine().flatMap { s in Int(s) } ?? 0
        if age < 20 {
            print("未成年なので購入できません。")
        }
    }

    func question2() {
        print("身長を入力してください。＞")
        let height = readLine()
                .flatMap { s in Float(s) }
                .map { f in f * 0.01 } ?? 1.0
        print("体重を入力してください。＞")
        let weight = readLine().flatMap { s in Float(s) } ?? 0.0
        let standard = height * height * 22.0
        print("あなたの標準体重は\(standard)です。")

        if weight > standard && (weight - standard) / standard * 100.0 > 14.0 {
            print("太り気味です。")
        } else if weight < standard && (weight - standard) / standard * 100.0 < -14.0 {
            print("痩せ気味です。")
        } else {
            print("普通ですね。")
        }
    }

    func question3() {
        let num = Int.random(in: 0..<100)
        print("０から９９の範囲の数値が決定されました。")
        print("決められた数値を予想し、この数値よりも大きな値を入力してください＞")
        let guess = readLine().flatMap { s in Int(s) } ?? 0
        print("決められた数値は\(num)です。")
        if guess > num {
            print("正解です。")
        } else {
            print("不正解です。")
        }
    }

    func question4() {
        let num = Int.random(in: 0..<100)
        print("０から９９の範囲の数値が決定されました。")
        print("決められた数値を予想し、この数値よりも大きな値を入力してください＞")
        let guess = readLine().flatMap { s in Int(s) } ?? 0
        print("決められた数値は\(num)です。")

        switch guess {
        case let x where x < 0 || x > 99:
            print("反則です！")
        case let x where x > num && x - num <= 10:
            print("大正解です！")
        case let x where x < num && num - x <= 10:
            print("惜しい！")
        case let x where x == num:
            print("お見事！")
        default:
            if guess > num {
                print("正解です。")
            } else {
                print("不正解です。")
            }
        }
    }
}

File: CSharpRewrite/CSharpRewrite/Assignments/K03.cs
using CSharpRewrite.Shared;

namespace CSharpRewrite.Assignments;

public class K03 : IQuestion
{
    public void Question1()
    {
        try
        {
            Console.Write("年齢を入力してください。＞");
            var age = int.Parse(Console.ReadLine() ?? "0");
            if (age < 20)
            {
                Console.WriteLine("未成年なので購入できません。");
            }
        }
        catch (Exception ex)
        {
            Console.Error.WriteLine(ex.Message);
        }
    }

    public void Question2()
    {
        try
        {
            Console.Write("身長を入力してください。＞");
            var height = double.Parse(Console.ReadLine() ?? "0.0");
            Console.Write("体重を入力してください。＞");
            var weight = double.Parse(Console.ReadLine() ?? "0.0");
            height /= 100.0;
            var standard = height * height * 22.0;
            Console.WriteLine("あなたの標準体重は" + standard + "です。");

            if (weight > standard && (weight - standard) / standard * 100 > 14)
            {
                Console.WriteLine("太り気味です。");
            }
            else if (weight < standard && (weight - standard) / standard * 100 < -14)
            {
                Console.WriteLine("痩せ気味です。");
            }
            else
            {
                Console.WriteLine("普通ですね。");
            }
        }
        catch (Exception ex)
        {
            Console.Error.WriteLine(ex.Message);
        }
    }

    public void Question3()
    {
        try
        {
            var randomNumber = new Random();
            var n = randomNumber.Next(0, 100);
            Console.WriteLine("０から９９の範囲の数値が決定されました。");
            Console.Write("決められた数値を予想し、この数値よりも大きな値を入力してください＞");
            var guess = int.Parse(Console.ReadLine() ?? "0");
            Console.WriteLine("決められた数値は" + n + "です。");

            Console.WriteLine(guess > n ? "正解です。" : "不正解です。");
        }
        catch (Exception ex)
        {
            Console.Error.WriteLine(ex.Message);
        }
    }

    public void Question4()
    {
        try
        {
            var randomNumber = new Random();
            var n = randomNumber.Next(0, 100);
            Console.WriteLine("０から９９の範囲の数値が決定されました。");
            Console.Write("決められた数値を予想し、この数値よりも大きな値を入力してください＞");
            var guess = int.Parse(Console.ReadLine() ?? "0");
            Console.WriteLine("決められた数値は" + n + "です。");
            
            Console.WriteLine(guess switch
            {
                > 100 or < 0 => "反則です！",
                var x when x > n && x - n <= 10 => "大正解！",
                var x when x < n && n - x <= 10 => "惜しい！",
                var x when x == n => "お見事！",
                _ => guess > n ? "正解です。" : "不正解です。"
            });
        }
        catch (Exception ex)
        {
            Console.Error.WriteLine(ex.Message);
        }
    }
}